In [1]:
%load_ext autoreload
%autoreload 2

import pandas as pd
import click

In [2]:
import build_database

In [159]:
pd.options.display.max_rows = 999
pd.options.display.max_columns = 999

In [3]:
DATABASE = "processed_data/locator_database.csv"

In [294]:
build_database.main()

Processing housing data sources...
Flagging problem landlords...
Computing transit access...
Building the database...
Saving the database...
Finished


In [23]:
db = pd.read_csv(DATABASE, index_col="index")

In [35]:
db.head()

Address  Monthly Rent Property Type  \
index                                                                           
1288108       1718 W 66th St 1, Chicago, IL 60636           800           Apt   
4012748  6130 S Eberhart Ave 1, Chicago, IL 60637          1200           Apt   
4017021     4827 S Seeley Ave , Chicago, IL 60609           600           Apt   
4036551        828 E 88th Pl 2, Chicago, IL 60619           875           Apt   
4036578     4620 S Evans Ave 1, Chicago, IL 60653          1150           Apt   

         Bath  Bed Voucher Necessary        Availability         Contact  \
index                                                                      
1288108   1.0  2.0               Yes  Check Availability             NaN   
4012748   2.0  3.0               Yes       Available Now  (773) 961-4367   
4017021   1.0  1.0               Yes  Check Availability             NaN   
4036551   1.0  3.0               Yes  Check Availability             NaN   
4036578   1.0  3.0               Yes  Check Availability             NaN   

                                                       URL        Lat  ...  \
index                                                                  ...   
1288108  http://chicagoha.gosection8.com/Section-8-hous...  41.774000  ...   
4012748  http://chicagoha.gosection8.com/Section-8-hous...  41.783000  ...   
4017021  http://chicagoha.gosection8.com/Section-8-hous...  41.805800  ...   
4036551  http://chicagoha.gosection8.com/Section-8-hous...  41.734190  ...   
4036578  http://chicagoha.gosection8.com/Section-8-hous...  41.810646  ...   

         2016_evict_rate 2016_evict_filing_rate  2011-2015_rent_perc_change  \
index                                                                         
1288108             1.54                   4.62                    0.002481   
4012748             2.81                   8.15                   -0.046617   
4017021             0.42                   2.09                    0.000000   
4036551             2.26                   7.89                    0.014996   
4036578             2.73                   5.45                    0.002683   

        2015-2019_rent_perc_change potential_bad_landlord  \
index                                                       
1288108                   0.113036                  False   
4012748                   0.026814                  False   
4017021                   0.083398                  False   
4036551                   0.097201                  False   
4036578                   0.030100                  False   

        bad_landlord_address num_stops_quart_mi  num_stops_half_mi  \
index                                                                
1288108                  NaN                  0                  2   
4012748                  NaN                  2                  4   
4017021                  NaN                  0                  2   
4036551                  NaN                  0                  0   
4036578                  NaN                  0                  0   

         num_stops_3quart_mi  num_stops_1_mi  
index                                         
1288108                    2               2  
4012748                    4               8  
4017021                    2               2  
4036551                    0               0  
4036578                    4               6  

[5 rows x 28 columns]

Address (str): full address. e.g. "1718 W 66th St 1, Chicago, IL 60636"
Monthly Rent (tuple of int): rent within (<min of range>, <max of range>)
Property Type (list of str): a list of the following options: 
                    "4-Plex", "Apt", "Duplex", "House", "Townhouse", "TriPlex"
Bath (tuple of float): number of bathrooms within (<min of range>, <max of range>)
Bed (tuple of float): number of bedrooms within (<min of range>, <max of range>)
Requires Voucher (bool): whether the unit requires HCV
Available Now (bool): whether the unit is available now
Neighborhood (str): a zillow neighborhood
L-Stop within _ Mile (float): a float of the following options: 
                              0.25, 0.5, 0.75, 1


In [163]:
DIST_TO_COL = {0.25: "num_stops_quart_mi",
                0.5: "num_stops_half_mi",
                0.75: "num_stops_3quart_mi",
                1: "num_stops_1_mi"}

In [236]:
c = {
    "Address": None,
    "Monthly Rent": (800, 1000),
    "Property Type": ["Apt"],
    "Bath": (1, 2),
    "Bed": (1, 1),
    "Available Now": True,
    "Neighborhood": None,
    "Has L-Stop within _ Mile": 0.25}

In [149]:
c

{'Address': None,
 'Monthly Rent': (800, 1000),
 'Property Type': ['Apt', 'Duplex'],
 'Bath': (1, 2),
 'Bed': None,
 'Available Now': True,
 'Neighborhood': 'Woodlawn',
 'Distance to nearest L-Stop': None}

In [166]:
rv = db

In [151]:
if c["Address"]:
    f1 = rv.Address == c["Address"]
    rv = rv[f1]

In [152]:
if c["Monthly Rent"]:
    lb_rent, ub_rent = c["Monthly Rent"]
    f2 = (rv["Monthly Rent"] >= lb_rent) & (rv["Monthly Rent"] <= ub_rent)
    rv = rv[f2]

In [153]:
if c["Property Type"]:
    f3 = rv["Property Type"].isin(c["Property Type"])
    rv = rv[f3]

In [154]:
if c["Bath"]:
    lb_bath, ub_bath = c["Bath"]
    f4 = (rv.Bath >= lb_bath) & (rv.Bath <= ub_bath)
    rv = rv[f4]

In [155]:
if c["Bed"]:
    lb_bed, ub_bed = c["Bath"]
    f5 = (rv.Bed >= lb_bed) & (rv.Bed <= ub_bed)
    rv = rv[f5]

In [156]:
if c["Available Now"]:
    f6 = rv.Availability == "Available Now"
    rv = rv[f6]

In [157]:
if c["Neighborhood"]:
    f7 = rv.Neighborhood == c["Neighborhood"]
    rv = rv[f7]

In [168]:
if c["Has L-Stop within _ Mile"]:
    d = c["Has L-Stop within _ Mile"]
    f8 = rv[DIST_TO_COL[d]] > 0
    rv = rv[f8]

In [171]:
rv

Address  Monthly Rent  \
index                                                                      
4012748           6130 S Eberhart Ave 1, Chicago, IL 60637          1200   
4062710             211 N Laporte Ave 2, Chicago, IL 60644           950   
4062846           6141 S Eberhart Ave 1, Chicago, IL 60637          1350   
4064659             6110 S Vernon Ave G, Chicago, IL 60637           900   
4076796              6110 S Vernon Ave , Chicago, IL 60637          1250   
4077614               3334 W Walnut St , Chicago, IL 60624          1150   
4079071               3334 W Walnut St , Chicago, IL 60624          1200   
4205319           3509 S Western Blvd 2, Chicago, IL 60609          1000   
4274287                646 E 62nd St  2, Chicago, IL 60637           997   
4283724      5317 W Ferdinand St 1st Fl, Chicago, IL 60644          1200   
4283730           5317 W Ferdinand St 2, Chicago, IL 60644          1400   
4287385            6158 S Langley Ave 2, Chicago, IL 60637          1090   
4290602            6350 S Langley Ave 2, Chicago, IL 60637          1100   
4302965              6448 S Peoria St 1, Chicago, IL 60621           900   
4318464          3953 W Van Buren St 13, Chicago, IL 60624           900   
4318488            3834 W Maypole Ave 1, Chicago, IL 60624           900   
4350143              6142 S Evans Ave 2, Chicago, IL 60637          1300   
4363485          7110 S Lafayette Ave 1, Chicago, IL 60621          1400   
4380165                6234 S Paulina 2, Chicago, IL 60636          1200   
4428509              6338 S Laflin St 2, Chicago, IL 60636          1300   
4444936                  4114 Gladys  1, Chicago, IL 60624          1200   
4445289             6142 S Evans Ave  3, Chicago, IL 60637          1300   
4446802              6214 S Evans Ave 3, Chicago, IL 60637          1500   
4466943            1836 South Harding 1, Chicago, IL 60623          1100   
4500012           4205 W Cullerton St 2, Chicago, IL 60623          1200   
4500334            4205 W Cullerton St , Chicago, IL 60623          1100   
4503354                 1420 W 64th St , Chicago, IL 60636           950   
4530053            5219 S Calumet Ave 1, Chicago, IL 60615          1250   
4530054            5219 S Calumet Ave 1, Chicago, IL 60615           900   
4530056             5219 S Calumet Ave , Chicago, IL 60615          1050   
4540144              210 N Wells St 404, Chicago, IL 60606          1800   
4540767             6428 S Vernon Ave 1, Chicago, IL 60637          1250   
4543108              7915 S Perry Ave 1, Chicago, IL 60620           850   
4543424          6449 S Langley Ave 203, Chicago, IL 60637           900   
4543495          536 N Leclaire Ave 208, Chicago, IL 60644          1200   
4543724            5610 S Indiana Ave 1, Chicago, IL 60637          1100   
4543732            5610 S Indiana Ave 1, Chicago, IL 60637          1100   
4543939               6417 S Evans Ave , Chicago, IL 60637          1150   
4544206            210 E Garfield Blvd , Chicago, IL 60615          1080   
4545701            4130 W Cullerton St , Chicago, IL 60623           880   
4545966            5113 S Indiana Ave 3, Chicago, IL 60615          1350   
4546137              3640 W Cermak Rd 8, Chicago, IL 60623          1250   
4548485          3846 W Congress Pkwy 2, Chicago, IL 60624           925   
4548955                 6158 S King Dr , Chicago, IL 60637          1100   
4550782              7044 S Perry Ave 2, Chicago, IL 60621          1100   
4551452           4711 W Lexington St 2, Chicago, IL 60644          1483   
4551545             3303 W Walnut St  1, Chicago, IL 60624          1200   
4551750          5840 N Sheridan Rd 506, Chicago, IL 60660           800   
4552033            6437 S Drexel Ave 2b, Chicago, IL 60637           977   
4552034          6215 S Langley Ave 301, Chicago, IL 60637           846   
4553227              1867 S Avers Ave  , Chicago, IL 60623          1300   
4555770            3144

In [172]:
import user

In [216]:
user.search(c, db, user.OUTPUT1)

2 search results saved in output/sample_output1.csv


In [181]:
ptypes = ["4-Plex", "Apt", "Duplex", "House", "Townhouse", "TriPlex"]

In [183]:
print(str(ptypes))

['4-Plex', 'Apt', 'Duplex', 'House', 'Townhouse', 'TriPlex']


In [203]:
dist_to_col = {0.25: "num_stops_quart_mi",
                0.5: "num_stops_half_mi",
                0.75: "num_stops_3quart_mi",
                1: "num_stops_1_mi"}

In [217]:
pd.read_csv(user.OUTPUT1)

index                                   Address  Monthly Rent  \
0  4530054   5219 S Calumet Ave 1, Chicago, IL 60615           900   
1  4771491  3900 W Van Buren St  , Chicago, IL 60624           950   

  Property Type  Bath  Bed Voucher Necessary   Availability         Contact  \
0           Apt   1.0  1.0               Yes  Available Now  (708) 473-5464   
1           Apt   1.0  1.0               Yes  Available Now  (312) 918-5955   

                                                 URL      Lat     Long  \
0  http://chicagoha.gosection8.com/Section-8-hous...  41.7997 -87.6172   
1  http://chicagoha.gosection8.com/Section-8-hous...  41.8755 -87.7233   

                          Coordinates         GEOID State County     City  \
0            POINT (-87.6172 41.7997)  170318361003    IL   Cook  Chicago   
1  POINT (-87.72329999999999 41.8755)  170312608002    IL   Cook  Chicago   

         Neighborhood  RegionID  2016_evict_rate  2016_evict_filing_rate  \
0     Washington Park    275927             4.60                   10.34   
1  East Garfield Park    269577             2.59                    7.49   

   2011-2015_rent_perc_change  2015-2019_rent_perc_change  \
0                   -0.033813                    0.038719   
1                   -0.012162                    0.034200   

   potential_bad_landlord  bad_landlord_address  num_stops_quart_mi  \
0                   False                   NaN                   2   
1                   False                   NaN                   2   

   num_stops_half_mi  num_stops_3quart_mi  num_stops_1_mi  
0                  4                    6               8  
1                  2                    6               8

In [245]:
cr = user.Criteria()

In [246]:
cr

Address: None
Monthly Rent: None
Property Type: None
Bath: None
Bed: None
Available Now: None
Neighborhood: None
Has L-Stop within _ Mile: None

In [301]:
cr.add_criteria(c)

In [293]:
cr.clear_criteria()

All search fields are now cleared


In [332]:
cr.update_criteria({'Neighborhood': ["Love"]})

In [334]:
user.search(cr, user.DB, user.SAMPLE_OUTPUT)

ValueError: Wrong input value for Neighborhood
Input a list containing the following: dict_keys(['Albany Park', 'Andersonville', 'Arcadia Terrace', 'Archer Heights', 'Ashburn', 'Avalon Park', 'Avondale', 'Back of the Yards', 'Belmont Central', 'Belmont Gardens', 'Belmont Heights', 'Belmont Terrace', 'Beverly', 'Beverly Woods', 'Big Oaks', 'Bowmanville', 'Brainerd', 'Bridgeport', 'Brighton Park', 'Bronzeville', 'Bucktown', 'Budlong Woods', 'Burnside', 'Cabrini Green', 'Calumet Heights', 'Canaryville', 'Chatham', 'Chicago Lawn', 'Chinatown', 'Chrysler Village', 'Clearing', 'Cottage Grove Heights', 'Cragin', 'Dearborn Park', 'Dunning', 'East Beverly', 'East Chatham', 'East Garfield Park', 'East Hyde Park', 'East Side', 'East Ukrainian Village', 'Edgebrook', 'Edgewater', 'Edgewater Glen', 'Edison Park', 'Englewood', 'Fernwood', 'Ford City', 'Forest Glen', 'Fulton River District', 'Gage Park', 'Galewood', 'Garfield Ridge', 'Gold Coast', 'Goose Island', 'Graceland West', 'Grand Crossing', 'Greektown', 'Gresham', 'Groveland Park', 'Heart of Chicago', 'Hegewisch', 'Hermosa', 'Hollywood Park', 'Humboldt Park', 'Hyde Park', 'Irving Park', 'Irving Woods', 'Jefferson Park', 'Kelvin Park', 'Kennedy Park', 'Kenwood', 'Kilbourn Park', 'Lake View', 'Lake View East', 'Lathrop Homes', 'LeClaire Courts', 'Lincoln Park', 'Lincoln Square', 'Little Village', 'Logan Square', 'Longwood Manor', 'Magnolia Glen', 'Marquette Park', 'Marynook', 'Mayfair', 'McKinley Park', 'Montclare', 'Morgan Park', 'Mount Greenwood', 'Near North', 'Near West Side', 'New Eastside', 'Noble Square', 'North Austin', 'North Center', 'North Kenwood', 'North Mayfair', 'North Park', 'Norwood Park East', 'Norwood Park West', "O'Hare", 'Oakland', 'Old Irving Park', 'Old Norwood Park', 'Old Town', 'Old Town Triangle', 'Oriole Park', 'Park Manor', 'Park West', 'Parkview', 'Pill Hill', 'Pilsen', 'Portage Park', 'Printers Row', 'Pulaski Park', 'Pullman', 'Ranch Triangle', 'Ravenswood', 'Ravenswood Manor', 'River North', 'River West', 'Rogers Park', 'Roscoe Village', 'Roseland', 'Rosemoor', 'Sauganash', 'Schorsch Forest View', 'Schorsch Village', 'Scottsdale', 'Sheffield Neighbors', 'South Austin', 'South Chicago', 'South Commons', 'South Deering', 'South Loop', 'South Shore', 'Stony Island Park', 'Streeterville', 'The Gap', 'The Loop', 'Tri-Taylor', 'Ukrainian Village', 'Union Ridge', 'University Village - Little Italy', 'Uptown', 'Vittum Park', 'Washington Heights', 'Washington Park', 'West Beverly', 'West Chatham', 'West Chesterfield', 'West De Paul', 'West Elsdon', 'West Englewood', 'West Garfield Park', 'West Humboldt Park', 'West Lawn', 'West Loop Gate', 'West Morgan Park', 'West Pullman', 'West Rogers Park', 'West Town', 'West Woodlawn', 'Wicker Park', 'Wildwood', 'Woodlawn', 'Wrightwood', 'Wrightwood Neighbors', 'Wrigleyville'])

In [327]:
cr.add_criteria({"Available Now": False})

In [319]:
d1 = db[(db["Monthly Rent"] >= 800)  & (db["Monthly Rent"] <= 1000)]

In [326]:
d1[d1["Neighborhood"] == "Woodlawn"]

Address  Monthly Rent  \
index                                                                
4274104     6637 S Greenwood Ave , Chicago, IL 60637           900   
4276320        6526 S Drexel Ave , Chicago, IL 60637          1000   
4284069    6530 S Greenwood Ave 2, Chicago, IL 60637           950   
4296404        6319 S Ellis Ave 2, Chicago, IL 60637          1000   
4552033      6437 S Drexel Ave 2b, Chicago, IL 60637           977   
4574061       925 E Marquette Rd , Chicago, IL 60637          1000   
4619162    6200 S University Ave , Chicago, IL 60637           911   
4641459     6619 S Greenwood Ave , Chicago, IL 60637          1000   
4647666     6619 S Greenwood Ave , Chicago, IL 60637           980   
4648350     6623 S Greenwood Ave , Chicago, IL 60637           950   
4682426  6649 S Blackstone Ave  2, Chicago, IL 60637          1000   
4693784      915 E Marquette Rd  , Chicago, IL 60637           975   
4728399   6619 S Greenwood Ave  2, Chicago, IL 60637          1000   
4759254    6625 S Greenwood Ave 2, Chicago, IL 60637          1000   
4804146           843 E 65th St 1, Chicago, IL 60637           900   
4804148           843 E 65th St 2, Chicago, IL 60637          1000   
4812680    6615 S Greenwood Ave 1, Chicago, IL 60637           894   
4812684    6615 S Greenwood Ave 3, Chicago, IL 60637           894   
4823114        6140 S Drexel Ave , Chicago, IL 60637           925   

        Property Type  Bath  Bed Voucher Necessary        Availability  \
index                                                                    
4274104           Apt   1.0  2.0               Yes       Available Now   
4276320           Apt   1.0  3.0               Yes       Available Now   
4284069           Apt   1.0  2.0               Yes  Check Availability   
4296404           Apt   1.0  3.0               Yes  Check Availability   
4552033           Apt   1.0  2.0               Yes  Check Availability   
4574061           Apt   1.0  2.0               Yes       Available Now   
4619162           Apt   1.5  3.0               Yes  Check Availability   
4641459           Apt   1.0  2.0               Yes       Available Now   
4647666           Apt   1.0  2.0               Yes       Available Now   
4648350           Apt   1.0  1.0               Yes  Check Availability   
4682426           Apt   1.0  2.0               Yes  Check Availability   
4693784           Apt   1.0  2.0               Yes  Check Availability   
4728399           Apt   1.0  2.0               Yes       Available Now   
4759254           Apt   1.0  2.0               Yes       Available Now   
4804146           Apt   1.0  2.0               Yes  Check Availability   
4804148           Apt   1.0  3.0               Yes  Check Availability   
4812680           Apt   1.0  3.0               Yes  Check Availability   
4812684           Apt   1.0  3.0               Yes  Check Availability   
4823114           Apt   1.5  1.0               Yes  Check Availability   

                Contact                                                URL  \
index                                                                        
4274104  (773) 961-4367  http://chicagoha.gosection8.com/Section-8-hous...   
4276320  (773) 961-4367  http://chicagoha.gosection8.com/Section-8-hous...   
4284069             NaN  http://chicagoha.gosection8.com/Section-8-hous...   
4296404             NaN  http://chicagoha.gosection8.com/Section-8-hous...   
4552033             NaN  http://chicagoha.gosection8.com/Section-8-hous...   
4574061  (773) 285-3310  http://chicagoha.gosection8.com/Section-8-hous...   
4619162             NaN  http://chicagoha.gosection8.com/Section-8-hous...   
4641459  (312) 852-3387  http://chicagoha.gosection8.com/Section-8-hous...   
4647666  (312) 852-3387  http://chicagoha.gosection8.com/Section-8-hous...   
4648350             NaN  http://chicagoha.gosection8.com/Section-8-hous...   
4682426             NaN  http://chicagoha.gosection8.com/Section-8-hous...   
46937

In [328]:
user.search(cr, db, user.OUTPUT1)

1 search results saved in output/sample_output1.csv


In [284]:
z = pd.read_csv("data/Neighborhood_Zri_AllHomesPlusMultifamily.csv", usecols=["City", "RegionName"])

In [281]:
zz = pd.read_csv( "processed_data/zillow_rindex_with_increase.csv")

In [285]:
z

RegionName              City
0                                      Northeast Dallas            Dallas
1                                              Maryvale           Phoenix
2                                       Upper West Side          New York
3                                     South Los Angeles       Los Angeles
4                                       Upper East Side          New York
5                                         Sunrise Manor         Las Vegas
6                                 Southeast Los Angeles       Los Angeles
7                                         East New York          New York
8                                       Paradise Valley           Phoenix
9                                    Washington Heights          New York
10                                          Deer Valley           Phoenix
11                                              Astoria          New York
12                                            Far North            Dallas
13                                   Bedford Stuyvesant          New York
14                                         Williamsburg          New York
15                                            Hollywood       Los Angeles
16                                       Camelback East           Phoenix
17                                            East Side           El Paso
18                                             Flatbush          New York
19                                     Southeast Dallas            Dallas
20                                             Alhambra           Phoenix
21                                        Crown Heights          New York
22                                              Jamaica          New York
23                                               Harlem          New York
24                                       Williamsbridge          New York
25                                          East Harlem          New York
26                                     North Scottsdale        Scottsdale
27                                            Roosevelt            Fresno
28                                       South Mountain           Phoenix
29                                        East San Jose          San Jose
30                                             Bushwick          New York
31                                        East Flatbush          New York
32                                           South Side    Corpus Christi
33                                                Alief           Houston
34                                               Corona          New York
35                      East Memphis-Colonial-Yorkshire           Memphis
36                                          Sunset Park          New York
37                                              Bullard            Fresno
38                                     Southwest Dallas            Dallas
39                                         Borough Park          New York
40                                          Bensonhurst          New York
41                                         Central City    Corpus Christi
42                                        West San Jose          San Jose
43                                            Northeast           El Paso
44                                             Elmhurst          New York
45                                                 East         Arlington
46                                            Koreatown       Los Angeles
47                                             Westlake       Los Angeles
48                                     South Scottsdale        Scottsdale
49                                       Blossom Valley          San Jose
50                                     North Long Beach        Long Beach
51                                       Sheepshead Bay          New York
52                                          South Bronx          New York
53                                            Sou

In [297]:
z[z.City == 'Chicago'].groupby('RegionName').groups

{'Albany Park': Int64Index([142], dtype='int64'),
 'Andersonville': Int64Index([4085], dtype='int64'),
 'Arcadia Terrace': Int64Index([2698], dtype='int64'),
 'Archer Heights': Int64Index([150], dtype='int64'),
 'Ashburn': Int64Index([904], dtype='int64'),
 'Avalon Park': Int64Index([3801], dtype='int64'),
 'Avondale': Int64Index([687], dtype='int64'),
 'Back of the Yards': Int64Index([289], dtype='int64'),
 'Belmont Central': Int64Index([357], dtype='int64'),
 'Belmont Gardens': Int64Index([1037], dtype='int64'),
 'Belmont Heights': Int64Index([1023], dtype='int64'),
 'Belmont Terrace': Int64Index([4316], dtype='int64'),
 'Beverly': Int64Index([924], dtype='int64'),
 'Beverly Woods': Int64Index([5794], dtype='int64'),
 'Big Oaks': Int64Index([2911], dtype='int64'),
 'Bowmanville': Int64Index([3348], dtype='int64'),
 'Brainerd': Int64Index([938], dtype='int64'),
 'Bridgeport': Int64Index([241], dtype='int64'),
 'Brighton Park': Int64Index([188], dtype='int64'),
 'Bronzeville': Int64Ind

In [278]:
l

['Albany Park',
 'Andersonville',
 'Arcadia Terrace',
 'Archer Heights',
 'Ashburn',
 'Avalon Park',
 'Avondale',
 'Back of the Yards',
 'Belmont Central',
 'Belmont Gardens',
 'Belmont Heights',
 'Belmont Terrace',
 'Beverly',
 'Beverly Woods',
 'Big Oaks',
 'Bowmanville',
 'Brainerd',
 'Bridgeport',
 'Brighton Park',
 'Bronzeville',
 'Bucktown',
 'Budlong Woods',
 'Burnside',
 'Cabrini Green',
 'Calumet Heights',
 'Canaryville',
 'Chatham',
 'Chicago Lawn',
 'Chinatown',
 'Chrysler Village',
 'Clearing',
 'Cottage Grove Heights',
 'Cragin',
 'Dearborn Park',
 'Dunning',
 'East Beverly',
 'East Chatham',
 'East Garfield Park',
 'East Hyde Park',
 'East Side',
 'East Ukrainian Village',
 'Edgebrook',
 'Edgewater',
 'Edgewater Glen',
 'Edison Park',
 'Englewood',
 'Fernwood',
 'Ford City',
 'Forest Glen',
 'Fulton River District',
 'Gage Park',
 'Galewood',
 'Garfield Ridge',
 'Gold Coast',
 'Goose Island',
 'Graceland West',
 'Grand Crossing',
 'Greektown',
 'Gresham',
 'Groveland Park